### PR TITLE
Refresh docs in view of iDynTree 2 release

### PR DESCRIPTION
**Pull request by traversaro**
There is a missing piece of changing the `KinDynComputationsWithEigen` to use the new interfaces that use `iDynTree::Span`  and `iDynTree::MatrixView`, but that will come in a following PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,67 +40,53 @@ but you can manually make sure that iDynTree searches or ignores a given depende
...
-##### Deprecated
-- [Kinematics and Dynamics Library](https://github.com/orocos/orocos_kinematics_dynamics)
-- [urdfdom](https://github.com/ros/urdfdom)
-- [irrlicht](http://irrlicht.sourceforge.net/)
-
 #### Install dependencies
 
-If you need to install also `YARP` and `ICUB`, it is recommended that you use the [`robotology-superbuild`](https://github.com/robotology/robotology-superbuild), but if you want to install them manually please
-check the documentation at [ICub Software Installation](http://wiki.icub.org/wiki/ICub_Software_Installation). The rest of the dependencies can be installed using standard operating system package managers.
+If you need to install also `YARP` and `ICUB`, it is recommended that you install iDynTree via the the [`robotology-superbuild`](https://github.com/robotology/robotology-superbuild). If instead you are not interested in the `YARP` and `ICUB` integration, you can easily install the rest of the dependencies using
```
Comment (S-Dafarra) [inline README.md:52]:
```suggestion
If you need to install also `YARP` and `ICUB`, it is recommended that you install iDynTree via the [`robotology-superbuild`](https://github.com/robotology/robotology-superbuild). If instead you are not interested in the `YARP` and `ICUB` integration, you can easily install the rest of the dependencies using
```

File: examples/cxx/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Copyright: 2020 Fondazione Istituto Italiano di Tecnologia
+# CopyPolicy: Released under the terms of the MIT license
+
+cmake_minimum_required(VERSION 3.16)
```
Comment (S-Dafarra) [inline examples/cxx/CMakeLists.txt:4]: This is required also in the parent folder.

File: examples/cxx/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+project(iDynTreeExamples)
```
Comment (S-Dafarra) [inline examples/cxx/CMakeLists.txt:6]: Isn't this a duplicate of the parent folder?

File: examples/cxx/CMakeLists.txt (hunk 2 above)
Comment (traversaro) [reply examples/cxx/CMakeLists.txt:4]: It is already there, unless I am missing something.

File: examples/cxx/CMakeLists.txt (hunk 3 above)
Comment (traversaro) [reply examples/cxx/CMakeLists.txt]: Yep, as a convenience I would like to be able to run cmake from both the examples directory and all the subdirectory. In any case I changed the project name at least.

File: examples/cxx/CMakeLists.txt (hunk 2 above)
Comment (S-Dafarra) [reply examples/cxx/CMakeLists.txt:4]: Sorry, I was indeed ambiguous. I meant that the same line was already present in the parent folder, so I was not sure whether this was needed.

Comment (traversaro) [reply examples/cxx/CMakeLists.txt:4]: I think that for examples it is convenient that each directory can be built on its own, starting from the leaf directories up to the most parent (used by tests).

Comment (S-Dafarra) [reply examples/cxx/CMakeLists.txt:4]: Ah, I see. Ok then

**Review by S-Dafarra — APPROVED** [review]
Btw, I just noticed that ``irrlicht`` is not listed as a dependency on Linux for the superbuild

**Comment by traversaro** [comment]
> 
> 
> Btw, I just noticed that `irrlicht` is not listed as a dependency on Linux for the superbuild

Good point. The big problem is that the visualizer does not work when loading unmodified URDFs and does not work with macOS, so for that was never fully announced/enforced as use.

**Pull request merged**